### PR TITLE
add(clientStore): try to speed up things

### PR DESCRIPTION
**Pull request by ybelMekk**
workflows:
* update runs-on

code:
* instead of looking up the clientId 2 times with find we bulk with both in same lookup, like a lazy lookup, if the audience is not present, the code fails like before.
* URL is deprecated, use URI().toURL()
* add findClients to clientRegistry
* add findClients to ClientStore, to map
* update upserQuery to only do update if there is changes to data

mock:
* update mocks

Tests
* tests new behaviour for storeClients
* add test for findClients
* simple benchmark, comparing find and findClients

consider:
validating audience input
cache of client data , either inmem or redis
cache of jwks is 6 hours, should it be longer?

### CODE REVIEW

**Comment by ybelMekk** [comment]
ikke bry om den detekt, skal fjerne det å bruke `jmailen.kotlinter` men tar det i en egen PR etter denne

File: src/test/kotlin/io/nais/security/oauth2/registration/ClientStoreTest.kt [hunk 1]
```diff
@@ -48,5 +54,68 @@ internal class ClientStoreTest {
...
+                findClients(listOf("testclient", "testclient2")) shouldBe mapOf(
+                    "testclient" to testclient,
+                    "testclient2" to testclient2
+                )
+            }
+        }
+    }
+
+    @Test
+    fun testFindClientsAndStorePerformance() {
```
Comment (tronghn) [inline src/test/kotlin/io/nais/security/oauth2/registration/ClientStoreTest.kt:74]: These tests might be flaky depending on the specific machine that the tests run on. I think it's fine for now, though we might want to re-assess if flakiness becomes an issue

**Review by tronghn — APPROVED** [review]
Some nitpicks, otherwise lgtm! 🎉

File: src/main/kotlin/io/nais/security/oauth2/routing/TokenExchangeApi.kt [hunk 2]
```diff
@@ -46,14 +47,28 @@ internal fun Routing.tokenExchangeApi(config: AppConfiguration) {
...
+                    clientFinder = { clientAssertionCredential ->
+                        val audience = params["audience"]
+                        val clientIds = listOfNotNull(audience, clientAssertionCredential.clientId).distinct()
+
+                        // Populate clientMap dynamically if not already populated
+                        if (clientMap.isEmpty()) {
+                            val fetchedClients = config.clientRegistry.findClients(clientIds)
+                            clientMap.putAll(fetchedClients)
+                        }
+                        clientMap[clientAssertionCredential.clientId]
```
[the remark this answers is not in the document]
Comment (ybelMekk) [reply src/main/kotlin/io/nais/security/oauth2/routing/TokenExchangeApi.kt:67]: agreed, i will give it a shot

**Pull request merged**